In [1]:
import numpy as np

# Likes matrix
likes = [
    [10, 20],
    [15, 25]
]

# Comments matrix
comments = [
    [2, 3],
    [4, 5]
]

# Matrix multiplication
result = np.dot(likes, comments)

print("Resulting Matrix:")
print(result)

Resulting Matrix:
[[100 130]
 [130 170]]


Two 2×2 matrices were created to represent Instagram likes and comments.

Using `np.dot()`, the two matrices were multiplied.

The resulting matrix is:

[[100, 130],
 [130, 170]]

Matrix multiplication is useful in machine learning because many calculations can be represented using matrices.

In [2]:
def linear_regression_predict(x, m, c):
    y = m * x + c
    return y


# Example values
x = 5
m = 4
c = 10

prediction = linear_regression_predict(x, m, c)

print("Predicted value:", prediction)

Predicted value: 30


The function `linear_regression_predict()` uses the linear regression equation:

y = mx + c

Here, x is the input, m is the slope, and c is the intercept.

For x = 5, m = 4, and c = 10:

y = (4 × 5) + 10 = 30

Therefore, the predicted output is **30**.

In [3]:
# Distances in kilometers
distances = [2, 5, 8]

# Linear regression values
m = 4
c = 10

# Calculate delivery time for each distance
for distance in distances:
    delivery_time = linear_regression_predict(distance, m, c)
    print(distance, "km =", delivery_time, "minutes")

2 km = 18 minutes
5 km = 30 minutes
8 km = 42 minutes


The linear regression equation is:

y = 4x + 10

Using this equation, the predicted delivery times are:

- 2 km → 18 minutes
- 5 km → 30 minutes
- 8 km → 42 minutes

This shows that delivery time increases as the distance increases.

In Flipkart's product price prediction, the **slope** shows how much the predicted product price changes when the product rating increases by 1 unit.

The **intercept** represents the predicted product price when the product rating is 0. Therefore, slope controls the change in price, while intercept represents the starting or base predicted price.

Matrix multiplication helps machine learning models process large amounts of user and song data.

In Spotify's recommendation system, user preferences and song features can be represented as matrices. Matrix multiplication can combine these matrices to calculate how well a song matches a user's interests.

The model can then use these results to recommend songs that the user is more likely to enjoy.